spectator-ext-jvm/src/main/java/com/netflix/spectator/jvm/Jmx.java

```diff
@@ -36,15 +35,51 @@ private Jmx() {
    * mbeans from the local jvm.
    */
   public static void registerStandardMXBeans(Registry registry) {
+    monitorClassLoadingMXBean(registry);
+    monitorThreadMXBean(registry);
+    monitorCompilationMXBean(registry);
+
     for (MemoryPoolMXBean mbean : ManagementFactory.getPlatformMXBeans(MemoryPoolMXBean.class)) {
       registry.register(new MemoryPoolMeter(registry, mbean));
     }
-
     for (BufferPoolMXBean mbean : ManagementFactory.getPlatformMXBeans(BufferPoolMXBean.class)) {
       registry.register(new BufferPoolMeter(registry, mbean));
     }
   }
 
+  private static void monitorClassLoadingMXBean(Registry registry) {
+    ClassLoadingMXBean classLoadingMXBean = ManagementFactory.getPlatformMXBean(ClassLoadingMXBean.class);
+    PolledMeter.using(registry)
+      .withName("jvm.classloading.classesLoaded")
+      .monitorValue(classLoadingMXBean, ClassLoadingMXBean::getLoadedClassCount);
+    PolledMeter.using(registry)
+      .withName("jvm.classloading.classesUnloaded")
+      .monitorValue(classLoadingMXBean, ClassLoadingMXBean::getUnloadedClassCount);
+  }
+
+  private static void monitorThreadMXBean(Registry registry) {
+    ThreadMXBean threadMXBean = ManagementFactory.getPlatformMXBean(ThreadMXBean.class);
+    PolledMeter.using(registry)
+      .withName("jvm.thread.startedThreadCount")
+      .monitorMonotonicCounter(threadMXBean, ThreadMXBean::getTotalStartedThreadCount);
+    PolledMeter.using(registry)
+      .withName("jvm.thread.threadCount")
+      .monitorValue(threadMXBean, ThreadMXBean::getThreadCount);
+    PolledMeter.using(registry)
+      .withName("jvm.thread.daemonThreadCount")
+      .monitorValue(threadMXBean, ThreadMXBean::getDaemonThreadCount);
+  }
+
+  private static void monitorCompilationMXBean(Registry registry) {
+    CompilationMXBean compilationMXBean = ManagementFactory.getPlatformMXBean(CompilationMXBean.class);
+    if (compilationMXBean.isCompilationTimeMonitoringSupported()) {
+      PolledMeter.using(registry)
+        .withName("jvm.compilation.compilationTime")
+        .withTag("compiler", compilationMXBean.getName())
+        .monitorMonotonicCounter(compilationMXBean, CompilationMXBean::getTotalCompilationTime);
```

Comment:
It's been a while since I worked with Spectator. Do I need a dimension tag here or should it not matter terribly?

Reply:
Not sure what you mean by making it a duration, do you mean have it report as a timer? As written it would report as a counter that would be `milliseconds/second` spent compiling. The unit should be converted to seconds before updating the counter as the convention is to always prefer base units. With the interface provided I don't think it can be reported as a timer since we cannot get individual samples.

Reply:
Oh right, yeah, timer is what I was thinking, and I'd forgotten how they're sampled.

I assume you mean something like:
```
      AtomicLong totalTime = new AtomicLong();
      PolledMeter.using(registry)
        .withName("jvm.compilation.compilationTime")
        .withTag("compiler", compilationMXBean.getName())
        .monitorValue(compilationMXBean, compilation -> {
          long currentTotalTime = compilation.getTotalCompilationTime();
          double compilationTime = (currentTotalTime - totalTime.get()) / 1000.0;
          totalTime.set(currentTotalTime);
          return compilationTime;
        });
```

Reply:
That will overwrite the gauge value with the last delta. I think the best way of handling this would be a monotonic counter that updates based on a double, see #956. Then it would just be:

```java
PolledMeter.using(registry)
        .withName("jvm.compilation.compilationTime")
        .withTag("compiler", compilationMXBean.getName())
        .monitorMonotonicCounterDouble(compilationMXBean, c -> c.getTotalCompilationTime() / 1000.0);
```

Reply:
Ah perfect, I'll rebase and switch back to monotonic counters.